examples/imports_function_env.rs

```diff
@@ -72,15 +71,15 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // We derive a default implementation of `WasmerEnv` here.
     #[derive(WasmerEnv, Clone)]
     struct Env {
-        counter: Arc<RefCell<i32>>,
+        counter: Arc<Mutex<i32>>,
```

Comment:
Why do we need to change the examples? The previous examples should work as is. Or am I missing something?

Reply:
Yeah, the previous examples were incorrect, we just didn't enforce it. `WasmerEnv` needs to be `Send` and `Sync` because `Instance` is `Send` and `Sync` and we don't synchronize access to functions. So it's possible in safe Rust to get a `NativeFunc` on multiple threads that points to the same underlying function.  In order for the `WasmerEnv` to be able to handle being used this way, it must be both `Send` and `Sync`.  This shouldn't affect much if any real code. For example, WASI was not affected at all by this change.  Emscripten was only affected because it was doing some weird stuff with pointers.